pkg/engine2/cli.go

```diff
@@ -256,22 +264,107 @@ func setupProfiling() func() {
 	return func() {}
 }
 
-func (em *EngineMain) RunEngine(cmd *cobra.Command, args []string) error {
+func (em *EngineMain) Run(context *EngineContext) (int, []engine_errs.EngineError) {
+	returnCode := 0
+	var engErrs []engine_errs.EngineError
+
+	zap.S().Info("Running engine")
+	err := em.Engine.Run(context)
+	if err != nil {
+		returnCode = 1
+		if ee, ok := err.(engine_errs.EngineError); ok {
+			engErrs = append(engErrs, ee)
+		} else {
+			engErrs = append(engErrs, engine_errs.InternalError{Err: engine_errs.ErrorsToTree(err)})
+		}
+	}
+
+	if len(context.Solutions) > 0 {
+		writeDebugGraphs(context.Solutions[0])
+		for _, d := range context.Solutions[0].GetDecisions().GetRecords() {
+			d, ok := d.(solution_context.MaybeErroDecision)
```

Comment:
what is MaybeErroDecision

Reply:
i see what it is before, just would rename since its not very clear

Reply:
It's mostly to interrogate the decision to see if it can be converted to an error. Originally I had like `ToError` but since it could return `nil` (for when a validation wasn't an error), I felt like it didn't match well. I'd take a suggestion, not sure how else to convey that "this decision _might_ be an error, call this method to find out"